compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/QueryExpressionTest.java

```diff
@@ -67,6 +67,11 @@ public void testQueryWithTableKeyword() {
         test("table key(name, age) from int a in b select e", "query-expr/query_expr_assert_11.json");
     }
 
+    @Test
+    public void testQueryWithMapKeyword() {
+        test("map from int a in b select e", "query-expr/query_expr_assert_71.json");
```

Comment:
```bal
function foo() {
    var x = map from int i in y select z; // [1] can parse

    map from int i in y select z; // [2] cannot parse.
}
```

We need to allow parsing [2] as well. This is for the create code action. The user tends to write the expression first and then use the code action to get the assigment.

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/36304/commits/5be02bc5633a5183fc99a80c4f1694540cb3ac64

Reply:
We need to add a test case to cover [2]. You can add test case under `ExpressionStatementTest`.